Universal/Sniffs/CodeAnalysis/MixedBooleanOperatorSniff.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * PHPCSExtra, a collection of sniffs and standards for use with PHP_CodeSniffer.
+ *
+ * @package   PHPCSExtra
+ * @copyright 2021 WoltLab GmbH, 2023 PHPCSExtra Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCSStandards/PHPCSExtra
+ */
+
+namespace PHPCSExtra\Universal\Sniffs\CodeAnalysis;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+/**
+ * Forbid mixing `&&` and `||` within a single expression without making precedence
+ * clear using parentheses.
+ *
+ * @link https://github.com/squizlabs/PHP_CodeSniffer/pull/3205
+ * @link https://github.com/php-fig/per-coding-style/issues/22
+ *
+ * @since 1.2.0
+ */
+final class MixedBooleanOperatorSniff implements Sniff
+{
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @since 1.2.0
+     *
+     * @return array<int|string>
+     */
+    public function register()
+    {
+        return [
+            \T_BOOLEAN_OR,
+            \T_BOOLEAN_AND,
+        ];
+    }
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @since 1.2.0
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+        $token  = $tokens[$stackPtr];
+
+        $start = $phpcsFile->findStartOfStatement($stackPtr);
```

Comment:
```suggestion
        $start = BCFile::findStartOfStatement($phpcsFile, $stackPtr);
```

(with `use PHPCSUtils\BackCompat\BCFile;`)

This safeguards that if/when that function would change upstream, the upstream change will be available on older PHPCS versions.